packages/aws-cdk/lib/cli/cli.ts

```diff
@@ -209,10 +213,6 @@ export async function exec(args: string[], synthesizer?: Synthesizer): Promise<n
       throw new ToolkitError('You must either specify a list of Stacks or the `--all` argument');
     }
 
-    if (args['telemetry-file'] && !configuration.settings.get(['unstable']).includes('telemetry')) {
```

Comment:
This is too late in the process because telemetry has already been initialized, and the error thrown is then considered as a telemetry event.

We should really structure the unstable protection mechanism better.